File: Problems.CSharp/Problems/0006-Sum of digits/Problem_6.cs
namespace Problems.CSharp.Problems._0006_Sum_of_digits
{
    public class Problem_6
    {
        public int ReadPositiveNumber(string Message)
        {
            int number;

            do
            {
                Console.Write(Message);
            } while (!int.TryParse(Console.ReadLine(), out number) || number <= 0);

            return number;
        }

        private List<int> TakeDigits(int number)
        {
            int remainder = 0;
            List<int> reversedNumbers = new List<int>();

            while (number > 0)
            {
                remainder = number % 10; // Take last digit from number
                number = number / 10; // Remove last digit from number

                reversedNumbers.Add(remainder);
            }

            return reversedNumbers;
        }

        public void SumDigits(int number)
        {
            int sum = 0;

            Console.WriteLine($"Sum digits of {number} is : ");

            foreach(int n in TakeDigits(number))
            {
                sum += n;
            }

            Console.Write(sum);
        }
    }
}

File: Problems.CPP/Problem_6.cpp
#include <iostream>
#include <vector>

using namespace std;

class Problem_6 {

private:

	vector<int> TakeDigits(int number)
	{
		vector<int> reversedNumber;
		int remainder = 0;

		while (number > 0)
		{
			remainder = number % 10; // extract the last digit from number
			number = number / 10; // Remove the last digit after extracting

			reversedNumber.push_back(remainder);
		}

		return reversedNumber;
	}

public:

	int ReadPositiveNumber(string Message) {

		int number;

		do {
			cout << Message;
			cin >> number;

		} while (number <= 0);

		return number;
	}

	void SumDigits(int number)
	{
		int sum = 0;

		cout << "\nSum digits of " << number << " is : " << endl;

		for (int n : TakeDigits(number))
		{
			sum += n;
		}

		cout << sum << endl;
	}
};